pkg/aws/eni/limits/limits.go

```diff
@@ -803,21 +823,45 @@ func populateStaticENILimits() {
 		"z1d.xlarge":        {Adapters: 4, IPv4: 15, IPv6: 15, HypervisorType: "nitro"},
 	}
 }
+func init() {
+	limits.m = make(map[string]ipamTypes.Limits)
+	limits.triggerDone = make(chan struct{})
+}
```

Comment:
nit: Given that the struct now has so many fields, I think it might be cleaner to introduce a proper struct with a constructor. I think that would also help with the suspected issues when unit tests are running in parallel.

My proposal would be: Let's introduce a regular struct here that with a regular constructor and no global state, e.g. 'limits.NewLimitsGetter()` or simliar which is constructed whenever we create a new `InstancesManager`. The `Node` object (which calls `limits.Get`) can then access this limits getter from there, rather than accessing a global variable here.

Reply:
I will also remove the https://github.com/cilium/cilium/blob/d0c8aec28bb7ad3ff5dc025b1641febe3cb3b286/pkg/aws/eni/limits/limits.go#L863-L877 related code and config since I can see people only use it for adding some unknown instance and not really overrding it unless you want me to reimplement it with this new struct and announce the deprecation first and remove it in 1.19